### PR TITLE
Simplify boolean expressions

### PR DESCRIPTION
**Pull request by thebrid**
## Description
Apply clang-tidy check `readability-simplify-boolean-expr`.

## Motivation and Context
Slight code simplification.

## Types of change

- [ ] **Bug fix** (non-breaking change which fixes an issue)
- [ ] **Clean up** (non-breaking change which removes non-working, unmaintained functionality)
- [ ] **Improvement** (non-breaking change which improves existing functionality)
- [ ] **New feature** (non-breaking change which adds functionality)
- [ ] **Breaking change** (fix or feature that will cause existing functionality to change)
- [ ] **Cosmetic change** (non-breaking change that doesn't touch code)
- [ ] **None of the above** (please explain below)

## Checklist:

- [ ] My code follows the **[Code Guidelines](https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md)** of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [ ] I have read the **[Contributing](https://github.com/xbmc/xbmc/blob/master/docs/CONTRIBUTING.md)** document
- [ ] I have added tests to cover my change
- [ ] All new and existing tests passed

### CODE REVIEW

File: xbmc/PasswordManager.cpp [hunk 1]
```diff
@@ -115,12 +115,9 @@ void CPasswordManager::SaveAuthenticatedURL(const CURL &url, bool saveToProfile)
 
 bool CPasswordManager::IsURLSupported(const CURL &url)
 {
-  if ( url.IsProtocol("smb")
-    || url.IsProtocol("nfs")
-    || url.IsProtocol("sftp"))
-    return true;
-
-  return false;
+  return url.IsProtocol("smb")
```
Comment (lrusak) [inline xbmc/PasswordManager.cpp]: @pkerling do we have any sort of code styling for multiple lines like this?

Comment (pkerling) [reply xbmc/PasswordManager.cpp]: Not as part of the guidelines, I think, but our `.clang-format` uses `ContinuationIndentWith: 4`. No aligning to the first line in any case.

Comment (thebrid) [reply xbmc/PasswordManager.cpp]: @lrusak & @pkerling updated now.

**Comment by ksooo** [comment]
jenkins build this please

**Review by ksooo — APPROVED** [review]

**Pull request merged**